src/server/scripts/Spells/spell_rogue.cpp

```diff
@@ -703,6 +706,120 @@ class spell_rog_tricks_of_the_trade_proc : public SpellScriptLoader
         }
 };
 
+// 51698,51700,51701 - Honor Among Thieves
+class spell_rog_honor_among_thieves : public SpellScriptLoader
+{
+public:
+    spell_rog_honor_among_thieves() : SpellScriptLoader("spell_rog_honor_among_thieves") { }
+
+    class spell_rog_honor_among_thieves_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_rog_honor_among_thieves_AuraScript);
+
+        bool CheckProc(ProcEventInfo& eventInfo)
+        {
+            Unit* caster = GetCaster();
+            if (!caster)
+                return false;
+
+            if (caster->GetTypeId() == TYPEID_PLAYER && !caster->GetSpellHistory()->HasCooldown(eventInfo.GetSpellInfo()->Id))
+                return true;
+
+            return false;
+        }
+
+        void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+        {
+            PreventDefaultAction();
+
+            Unit* caster = GetCaster();
+            if (!caster)
+                return;
+
+            Unit* target = GetTarget();
+            target->CastSpell(target, eventInfo.GetSpellInfo()->Id, true, nullptr, aurEff, caster->GetGUID());
```

Comment:
thats wrong, eventInfo.GetSpellInfo() return the spell which triggers the aura proc